### ISSUE TITLE
[Bug] type check for vector does not take into account vector of vector

### ISSUE THREAD

**Issue by dariorussi** [libra/libra#3277]
# 🐛 Bug
the following code shows the problem
```
import 0x0.Vector;

main() {
    let other_vec: vector<vector<u8>>;
    other_vec = Vector.empty<vector<u8>>();
    assert(Vector.is_empty<vector<u8>>(&other_vec), 2);
    return;
}
```
it seems 
https://github.com/libra/libra/blob/master/language/move-vm/types/src/values/values_impl.rs#L1420-L1430 
is missing the check for the type being a `Vector` and it ends in the error arm of the match.
Same on all other vector APIs. Let's make sure we have tests as well. 

Reported here: https://community.libra.org/t/push-back-on-a-vector-with-other-vectors/2706

**Comment by sblackshear** [comment]
```
let digests = Vector::empty();
Vector::push_back(&mut digests, x"00")
```
is a small Move source example that repros FWIW

**Comment by sblackshear** [comment]
Fixed by 2c07e61b5da4d59612c63937fd099ca6e5eb47e0, thanks Victor!!
